src/jrd/tra.cpp

```diff
@@ -3651,11 +3652,45 @@ void jrd_tra::checkBlob(thread_db* tdbb, const bid* blob_id)
 			if (blb_relation->rel_security_name.isEmpty())
 				MET_scan_relation(tdbb, blb_relation);
 			SecurityClass* s_class = SCL_get_class(tdbb, blb_relation->rel_security_name.c_str());
-			if (s_class && !s_class->scl_blb_access)
+			if (s_class)
 			{
-				SCL_check_access(tdbb, s_class, 0, 0, NULL, SCL_select, SCL_object_table, false, 
-					blb_relation->rel_name);
-				s_class->scl_blb_access = true;
+				if (s_class->scl_blb_access == SecurityClass::BA_UNKNOWN)
+				{
+					// Relation has not been checked for access rights
+					try
+					{
+						SCL_check_access(tdbb, s_class, 0, 0, NULL, SCL_select, SCL_object_table, false,
+							blb_relation->rel_name);
+						s_class->scl_blb_access = SecurityClass::BA_SUCCESS;
+					}
+					catch (const Exception&)
+					{
+						if (tdbb->tdbb_status_vector->getErrors()[1] != isc_no_priv)
+							throw;
+
```

Comment:
Should we check catched exception here, not tdbb_status_vector ?

Reply:
OK, I will fix it.

Reply:
Thanks, but it seems not enough - few lines below you clear tdbb_status_vector.
It could clear some previous error\warning.
Usually, if code handle exception and not re-throw it, it set temporary status-vector to not overwrite tdbb's one.